### PR TITLE
Store information about each proxy in ModuleExtensionUsage

### PR DESCRIPTION
**Pull request by Wyverald**
The current ModuleExtensionUsage class was designed in the days before we allowed multiple `use_extension` calls for the same extension in the same module. This PR brings the class up to speed to better reflect the actual API.

This will also lay the groundwork for `bazel mod tidy` working with `include()`d segments.

Work towards https://github.com/bazelbuild/bazel/issues/22063

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java [hunk 1]
```diff
@@ -445,13 +445,14 @@ public static RootModuleFileValue evaluateRootModuleFile(
       throw errorf(Code.BAD_MODULE, "error executing MODULE.bazel file for the root module");
     }
     for (ModuleExtensionUsage usage : module.getExtensionUsages()) {
-      if (usage.getIsolationKey().isPresent() && usage.getImports().isEmpty()) {
+      ModuleExtensionUsage.Proxy firstProxy = usage.getProxies().getFirst();
```
Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java:448]: Maybe call this `onlyProxy` for clarity?

File: src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java [hunk 2]
```diff
@@ -189,36 +169,32 @@ void addImport(
...
-              .setLocation(location)
-              .setImports(ImmutableBiMap.copyOf(imports))
-              .setDevImports(devImports.build())
-              .setHasDevUseExtension(hasDevUseExtension)
-              .setHasNonDevUseExtension(hasNonDevUseExtension)
+              .setProxies(proxies)
               .setTags(tags.build());
       if (isolate) {
-        if (exportedName == null) {
+        ModuleExtensionUsage.Proxy firstProxy = proxies.getFirst();
```
Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java:188]: Also rename to `onlyProxy` and use `Iterables.getOnlyElement`?

**Review by fmeum — APPROVED** [review]
Much nicer!

File: src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java (hunk 1 above)
Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java:448]: Maybe move this into the `if` and rename to `onlyProxy`?

Comment (Wyverald) [reply src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java:448]: the `if` itself contains a reference to `firstProxy`.

File: src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java (hunk 2 above)
Comment (Wyverald) [reply src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java]: done

**Review by fmeum — APPROVED** [review]

**Pull request closed without merging**